spec/README.mediawiki

```diff
@@ -163,6 +163,17 @@ field will be non-<code>null</code> and the <code>result</code> field will be
 <code>null</code>.
 The opposite is true in the case of a success.
 
+To accommodate bi-directional communications over a single connection,  JSON-RPC
+calls are wrapped in a <code>Message</code> object that specifies a message
+<code>type</code>.
+
+<pre>
+{
+	"type": int, // 1: response, 2: request
+	"payload": {...} // JSON-RPC request or response
```

Comment:
Shouldn't the payload be the serialized bytes of the request/response?

Reply:
That's an implementation detail. This is the actual JSON structure. 

In go, we unmarshal the `payload` to bytes at first until we can read the `type`, but not every language will need that step.

Reply:
Even in go we're not really dealing with a binary serialization of this data.  Although `json.RawMessage` is `[]byte`, there's no binary serialization there -- you can see the JSON request/response just by `string(payload)`

Reply:
Do we anticipate that every message can be classified as either a request or response?  A request demands a response, but what about other unsolicited messages?

Reply:
> Do we anticipate that every message can be classified as either a request or response? A request demands a response, but what about other unsolicited messages?

you're right, https://github.com/decred/dcrdex/tree/master/spec#progress-notifications would need a notification type for the wrapper.